cmake/test/gtest.cmake

```diff
@@ -85,97 +85,261 @@ function(catkin_add_executable_with_gtest target)
+
+    get_filename_component(INCLUDE_DIR ${_GMOCK_INCLUDES} PATH)
+    get_filename_component(INCLUDE_DIR ${INCLUDE_DIR} PATH)
+
+    set(${found} TRUE PARENT_SCOPE)
+    set(${base_dir} ${BASE_DIR} PARENT_SCOPE)
+    set(${include_dir} ${gtest_include_dir} ${INCLUDE_DIR} PARENT_SCOPE)
+    set(${lib_dir} ${CMAKE_BINARY_DIR}/gmock PARENT_SCOPE)
+    set(${libs} "gmock" PARENT_SCOPE)
+    set(${main_libs} "gmock_main" PARENT_SCOPE)
+  endif()
+endfunction()
+
+find_package(GMock QUIET)
+if(NOT GMOCK_FOUND)
+  find_package(GTest QUIET)
+  if(NOT GTEST_FOUND)
+    # only add gmock/gtest directory once per workspace
+    if(NOT TARGET gtest AND NOT TARGET gmock)
+      # Fall back to system-installed gmock source (e.g. Ubuntu)
+      set(_include_paths "/usr/include/gmock")
       if(CATKIN_TOPLEVEL)
-        message(STATUS "gtest not found, C++ tests can not be built. Please install the gtest headers globally in your system or checkout gtest (by running 'svn checkout http://googletest.googlecode.com/svn/tags/release-1.6.0 gtest' in the source space '${CMAKE_SOURCE_DIR}' of your workspace) to enable gtests")
-      else()
-        message(STATUS "gtest not found, C++ tests can not be built. Please install the gtest headers globally in your system to enable gtests")
+        # Ensure current workspace is searched before system path
+        list(INSERT _include_paths 0 "${CMAKE_SOURCE_DIR}/gmock/include/gmock")
+      endif()
+
+      set(_source_paths "/usr/src/gmock/src")
+      if(CATKIN_TOPLEVEL)
+        # Ensure current workspace is searched before system path
+        list(INSERT _source_paths 0 "${CMAKE_SOURCE_DIR}/gmock/src")
+      endif()
+
+      catkin_find_gmock_source("${_include_paths}" "${_source_paths}" gmock_found
+                               gmock_base_dir gmock_include_dir gmock_lib_dir
+                               gmock_libs gmock_main_libs)
+
+      # If we found gmock, set it up to be built (which will also build gtest,
+      # since it's bundled)
+      if(gmock_found)
+        set(GMOCK_FROM_SOURCE_FOUND ${gmock_found} CACHE INTERNAL "")
+        set(GMOCK_FROM_SOURCE_INCLUDE_DIRS ${gmock_include_dir} CACHE INTERNAL "")
+        set(GMOCK_FROM_SOURCE_LIBRARY_DIRS ${gmock_lib_dir} CACHE INTERNAL "")
+        set(GMOCK_FROM_SOURCE_LIBRARIES ${gmock_libs} CACHE INTERNAL "")
+        set(GMOCK_FROM_SOURCE_MAIN_LIBRARIES ${gmock_main_libs} CACHE INTERNAL "")
+
+        set(_CATKIN_SKIP_INSTALL_RULES TRUE)
+        function(install)
+          if(_CATKIN_SKIP_INSTALL_RULES)
+            return()
+          endif()
+          _install(${ARGN})
+        endfunction()
+        add_subdirectory(${gmock_base_dir} ${gmock_lib_dir})
+        set(_CATKIN_SKIP_INSTALL_RULES FALSE)
+
+        set_target_properties(${gmock_libs} ${gmock_main_libs}
+                              PROPERTIES EXCLUDE_FROM_ALL 1)
+
+        message(STATUS "Found gmock sources under '${gmock_base_dir}': gtests will be built")
```

Comment:
The message seems inconsistent. The first part mentions gmock - the second part gtest.

Reply:
Fixed in 1d5e21c, when gmock is found and built, so is gtests.